src/CoreBundle/Resources/public/javascript/navigationTree.js

```diff
@@ -206,35 +226,51 @@ $(document).ready(function () {
                 "drag_selection": false
             }
         })
-        .on("move_node.jstree", function (e, data) {
-            moveNode(data.node.id, data.parent, data.position);
-        })
         .on("select_node.jstree", function (e, data) {
             connectPageOnSelect(data.node.id);
         })
         .on("deselect_node.jstree", function (e, data) {
             disconnectPageOnDeselect(data.node.id);
         });
-
 });
 
-/*
- * TreeNode field: sets selected node
- */
+function updateCurrentPageOrPortal(newNodeId) {
+    let url = navTreeDataContainer.data('update-selection-url') + '&sRestriction=' + newNodeId + '&nodeId=' + newNodeId;
+    GetAjaxCallTransparent(url, updateCurrentPageOrPortalSuccess);
+}
+
+function updateCurrentPageOrPortalSuccess(nodeId, responseMessage) {
+    if ('success' === responseMessage) {
+        const fieldName = navTreeDataContainer.data('field-name');
+        chooseTreeNode(fieldName, nodeId);
+    }
+}
+
 function chooseTreeNode(fieldName, newId) {
     parent.$('#' + fieldName).val(newId);
-    var newPath = $('#' + fieldName + '_tmp_path_' + newId).html();
+    let newPath = "";
```

Comment:
is let (and const) already used in other locations (in the core js)?

(my phpstorm warns that these language features might be too new..)

Reply:
I searched for it. Only within coreui, but tthere very often.

Reply:
We can use "let" everywhere in backend code.